components/scream/src/physics/share/physics_saturation_impl.hpp

```diff
@@ -84,12 +84,12 @@ Functions<S,D>::MurphyKoop_svp(const Spack& t_atm, const bool ice, const Smask&
 template <typename S, typename D>
 KOKKOS_FUNCTION
 typename Functions<S,D>::Spack
-Functions<S,D>::polysvp1(const Spack& t, const bool ice, const Smask& range_mask)
+Functions<S,D>::polysvp1(const Spack& t, const bool ice, const Smask& range_mask, const char* caller)
 {
   // REPLACE GOFF-GRATCH WITH FASTER FORMULATION FROM FLATAU ET AL. 1992, TABLE 4 (RIGHT-HAND COLUMN)
 
   //First check if the temperature is legitimate or not
-  check_temperature(t, "polysvp1", range_mask);
+  check_temperature(t, ekat::impl::strlen(caller) ? "polysvp1" : caller, range_mask);
```

Comment:
Same here re: `caller`. Note that this is a function call, so every call to MurphyKoop_svp will encounter this line, and the args to `check_temperature` will be evaluated always. So we don't want to call `strlen` every time.